src/agent/main.cpp

```diff
@@ -230,7 +232,11 @@ void otPlatReset(otInstance *aInstance)
 
     gPlatResetReason = OT_PLAT_RESET_REASON_SOFTWARE;
 
-    otSysDeinit();
+    if (gApp)
```

Comment:
Maybe we can use singleton pattern like otbr::Application::Get() ?

Thoughts? @wgtdkp @superwhd

Reply:
I don't see big benefits of making it a singleton. The `Application` class is not used across a lot of files. Also, we're not  reusing the same `Application` instance when doing a factory reset. Instead, we need to de-init the old instance and create a new instance, and assign the global instance as the new instance.